tools/state-viewer/src/cli.rs

```diff
@@ -257,6 +270,10 @@ pub struct ApplyRangeCmd {
     only_contracts: bool,
     #[clap(long)]
     use_flat_storage: bool,
+    /// Use the data stored in trie, but without paying extra gas costs.
+    /// This could be used to simulate flat storage when the latter is not present.
+    #[clap(long)]
+    use_trie_for_free: bool,
```

Comment:
This is just fine as is but I wonder if we could auto detect the right configuration. It should be possible to check
a) if flat storage should be used, based on the protocol version of the block
b) if flat storage is present at the given height

Then you can setup the db source based on that. This is totally optional as in practice we will almost always use (false, true).

Reply:
Interesting idea, maybe for improvement as a default it could use autodetect, I'll file an issue. I feel there are interesting scenarios (to check some stuff) where someone would prefer to fine tune the flags, so it makes sense to leave such option available

Reply:
Yeah in the follow up you could have something like `enum StorageSource { AutoDetect, Trie, TrieFree, FS }`